### DISCUSSION TITLE
0.5.0 edge case

### DISCUSSION THREAD

**Discussion by james-d-elliott** [wneessen/go-mail#328] [General]
There's an edge case where performing a no auth SMTP connection on 0.5.0 has changed. Specifically to be able to do this (SMTP without auth) now you must explicitly set `mail.WithSMTPAuthCustom(nil)`  or `mail.WithSMTPAuth(mail.SMTPAuthCustom)` with no other auth option in order to perform this action, while prior if no auth option was ever used it was happy to perform no auth.

While this is technically a breaking change I think it's overall a good move for the lib and makes the interaction more explicit. I suspect the original behavior was never intended.

**Comment by wneessen** [comment]
Thanks for bringing this up James. Do you have a code example for this behaviour? Maybe it's useful to add it as a test, so we don't accidently breakt things with a next release.

**Comment by wneessen** [comment]
Linking #332 here for completeness.
